lib/models/finamp_models.dart

```diff
@@ -1209,20 +1221,24 @@ enum DownloadItemStatus {
 /// The type of a BaseItemDto as determined from its type field.
 /// Enumerated by Isar, do not modify order or delete existing entries
 enum BaseItemDtoType {
-  unknown(null, false),
-  album("MusicAlbum", false),
-  artist("MusicArtist", true),
-  playlist("Playlist", true),
-  genre("MusicGenre", true),
-  song("Audio", false),
-  library("CollectionFolder", true),
-  folder("Folder", false),
-  musicVideo("MusicVideo", false);
-
-  const BaseItemDtoType(this.idString, this.expectChanges);
+  unknown(null, true, null),
+  album("MusicAlbum", false, [song]),
+  artist("MusicArtist", true, [album, song]),
+  playlist("Playlist", true, [song]),
+  genre("MusicGenre", true, [album, song]),
+  song("Audio", false, []),
+  library("CollectionFolder", true, [album, song]),
+  folder("Folder", true, null),
+  musicVideo("MusicVideo", false, []);
+
+  const BaseItemDtoType(this.idString, this.expectChanges, this.childTypes);
 
   final String? idString;
   final bool expectChanges;
+  final List<BaseItemDtoType>? childTypes;
+
+  bool get expectChangesInChildren =>
+      childTypes?.any((x) => x.expectChanges) ?? true;
```

Comment:
Right now everything that has children of type album also has songs, which makes sense. I'm wondering if this could be done implicitly, where we check if the child type also has children that might change.  
This is just an idea to make supporting downloads of collections in the future easier, which can contain basically and child type. I am however not sure if this would make sense wrt the code, because I don't know if there's any benefit in recursively modelling the contained children and their types, or if just using the type `folder` (which always expects changes) is enough already.

Reply:
The only place this list is used right now is the expectChangesInChildren bool.  I almost just hardwired in all the values for that instead of this list, and I might still change to that method, because calculating them like this sort of shows the logic, but sort of doesn't.  It's currently false for every type of collection except folder, which we don't even use, so it's a bit speculative.